### PR TITLE
Add three missing required dependencies in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
There could still be missing dependencies, I'm not using a fresh environement
to test that. Also added the specific required git version see
https://github.com/git/git/blob/master/Documentation/RelNotes/1.8.2.txt

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,12 +4,16 @@
 
 - The complete test suite depends on having at least the following installed
   (possibly not a complete list)
-  - git (A sufficiently newer version is required to run pre-push tests)
+  - git (Version 1.8.2 or above is required to run pre-push tests)
```
Comment (asottile) [inline CONTRIBUTING.md:7]: there's also a version needed to run the `pre-merge-commit` tests

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,12 +4,16 @@
...
   - python2 (Required by a test which checks different python versions)
   - python3 (Required by a test which checks different python versions)
   - tox (or virtualenv)
   - ruby + gem
   - docker
+  - conda
+  - cargo (required by tests for rust dependencies)
+  - go (required by tests for go dependencies)
```
Comment (asottile) [inline CONTRIBUTING.md:15]: there's also a requirement for swift

File: CONTRIBUTING.md (hunk 1 above)
Comment (Pierre-Sassoulas) [reply CONTRIBUTING.md]: OK, I changed this to 2.24.0.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Pierre-Sassoulas) [reply CONTRIBUTING.md:15]: Added in latest commit.

**Pull request merged**